### PR TITLE
First stab at implementing logic for new des api stub. Plenty of warn…

### PR DESCRIPTION
**Pull request by mattrobertsky**
…ings to resolve and probably lots to strip out but committing for review WIP (PLA-1575)

@nkaraolis please review.

### CODE REVIEW

File: app/uk/gov/hmrc/pla/des/stub/MicroserviceHelloWorld.scala [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/*
```
Comment (nkaraolis) [inline app/uk/gov/hmrc/pla/des/stub/MicroserviceHelloWorld.scala]: Remove this file

File: app/uk/gov/hmrc/pla/des/stub/controllers/PLAStubController.scala [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    *
+    * @param ltaRef short for lifetimeAllowanceReference
+    * @return
+    */
+  private def ltaRefValidator(ltaRef: String): Boolean = {
+    ltaRef.matches("^(IP14|IP16|FP16)[0-9]{10}[ABCDEFGHJKLMNPRSTXYZ]$") | ltaRef.matches("^[1-9A][0-9]{6}[ABCDEFHXJKLMNYPQRSTZW]$")
+  }
+}
+
+object ControllerHelper {
```
Comment (nkaraolis) [inline app/uk/gov/hmrc/pla/des/stub/controllers/PLAStubController.scala]: Shouldn't need this object as we check for the headers using our own method

File: app/uk/gov/hmrc/pla/des/stub/model/CreateLTAProtectionRequest.scala [hunk 3]
```diff
@@ -0,0 +1,48 @@
+/*
```
Comment (nkaraolis) [inline app/uk/gov/hmrc/pla/des/stub/model/CreateLTAProtectionRequest.scala]: Remove this file

File: app/uk/gov/hmrc/pla/des/stub/model/PensionDebit.scala [hunk 4]
```diff
@@ -0,0 +1,25 @@
+/*
```
Comment (nkaraolis) [inline app/uk/gov/hmrc/pla/des/stub/model/PensionDebit.scala]: Remove this file

File: app/uk/gov/hmrc/pla/des/stub/model/Protection.scala [hunk 5]
```diff
@@ -0,0 +1,120 @@
+/*
```
Comment (nkaraolis) [inline app/uk/gov/hmrc/pla/des/stub/model/Protection.scala]: Remove this file

File: test/uk/gov/hmrc/pla/des/stub/controllers/ControllerSpec.scala [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+    "return a failure with three validation errors" in {
+      val requestJs = ControllerHelper.addExtraRequestHeaderChecks(TestData.emptyHeadersExample, TestData.invalidRequestJsError)
+      requestJs.isSuccess shouldBe false
+      println("RESULT ==> " + requestJs)
+      requestJs.asInstanceOf[JsError].errors.seq.size shouldBe 1
+      requestJs.asInstanceOf[JsError].errors.seq.head._2.size shouldBe 3
+    }
+  }
+
+  "PSA Lookup" should {
```
Comment (nkaraolis) [inline test/uk/gov/hmrc/pla/des/stub/controllers/ControllerSpec.scala]: Keep the PSA Lookup stuff and get rid of the rest

File: test/uk/gov/hmrc/pla/des/stub/model/ModelSpec.scala [hunk 7]
```diff
@@ -0,0 +1,320 @@
+/*
```
Comment (nkaraolis) [inline test/uk/gov/hmrc/pla/des/stub/model/ModelSpec.scala]: Remove this file and the rules spec as well

File: app/uk/gov/hmrc/pla/des/stub/model/PSALookupUpdatedResult.scala [hunk 8]
```diff
@@ -0,0 +1,28 @@
+/*
```
Comment (nkaraolis) [inline app/uk/gov/hmrc/pla/des/stub/model/PSALookupUpdatedResult.scala]: Rename to PSALookupResult

File: app/uk/gov/hmrc/pla/des/stub/MicroserviceHelloWorld.scala (hunk 1 above)
Comment (mattrobertsky) [inline app/uk/gov/hmrc/pla/des/stub/MicroserviceHelloWorld.scala]: done

File: app/uk/gov/hmrc/pla/des/stub/controllers/PLAStubController.scala (hunk 2 above)
Comment (mattrobertsky) [inline app/uk/gov/hmrc/pla/des/stub/controllers/PLAStubController.scala]: done

**Comment by mattrobertsky** [comment]
Changes done, please review.

**Pull request merged**